MediaManager/Plugin.MediaManager.Abstractions/Implementations/PlaybackController.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Threading.Tasks;
+using Plugin.MediaManager.Abstractions.Enums;
+
+namespace Plugin.MediaManager.Abstractions.Implementations
+{
+    public class PlaybackController: IPlaybackController
+    {
+        private readonly IMediaManager _mediaManager;
+
+        private const int StepSeconds = 10;
+
+        private IMediaQueue Queue => _mediaManager.MediaQueue;
+
+        private double PositionSeconds => _mediaManager.Position.TotalSeconds;
+
+        public PlaybackController(IMediaManager mediaManager)
+        {
+            _mediaManager = mediaManager;
+        }
+
+        public async Task PlayPause()
+        {
+            var status = _mediaManager.Status;
+
+            var isPaused = status == MediaPlayerStatus.Paused;
+            var isStopped = status == MediaPlayerStatus.Stopped;
+
+            if (isPaused || isStopped)
+            {
+                await Play();
+            }
+            else
+            {
+                await Pause();
+            }
+        }
+
+        public async Task Play()
+        {
+            await _mediaManager.Play();
+        }
+
+        public async Task Pause()
+        {
+            await _mediaManager.Pause();
+        }
+
+        public async Task Stop()
+        {
+            await _mediaManager.Stop();
+        }
+
+        public async Task PlayPreviousOrSeekToStart()
+        {
+            if (PositionSeconds > 3)
+            {
+                await SeekToStart();
+            }
+            else
+            {
+                await PlayPrevious();
+            }
+        }
+
+        public async Task PlayNext()
+        {
+            await _mediaManager.PlayNext();
+        }
+
+        public async Task PlayPrevious()
+        {
+            if (Queue.HasPrevious())
+            {
+                await _mediaManager.PlayPrevious();
+            }
+            else
+            {
+                await SeekToStart();
+            }
+        }
+
+        public async Task SeekToStart()
+        {
+            await SeekTo(0);
+        }
+
+        public async Task StepForward()
+        {
+            var destination = PositionSeconds + StepSeconds;
+
+            await SeekTo(destination);
+        }
+
+        public async Task StepBackward()
+        {
+            var destination = PositionSeconds - StepSeconds;
+
+            await SeekTo(destination);
+        }
+
+        public async Task SeekTo(double seconds)
+        {
+            if (_mediaManager.Duration.TotalSeconds < seconds)
+            {
+                seconds = _mediaManager.Duration.TotalSeconds;
```

Comment:
The `IPlaybackManager.Duration` property is of type TimeSpan and it indicates the total duration of the Current Media File.